src/uwtools/tests/drivers/test_driver.py

```diff
@@ -83,13 +101,160 @@ def schema():
+    with patch.object(Driver, "_create_run_directory") as _create_run_directory:
+        Driver.create_directory_structure(run_dir)
+    _create_run_directory.assert_called_once_with(run_dir, "delete")
+
+
+def test_create_directory_structure_bad_existing_act():
+    with raises(ValueError):
+        Driver.create_directory_structure(run_directory="/some/path", exist_act="foo")
+
+
+def test_run_cmd_expected(mpi_config, tmp_path):
+    config_file = tmp_path / "config.yaml"
+    YAMLConfig.dump_dict(config_file, mpi_config)
+    with patch.object(Driver, "_validate", return_value=True):
+        driver = ConcreteDriver(config_file=config_file)
+    driver._config = driver._experiment_config["component"]
+    assert driver.run_cmd() == "srun bar foo"
+
+
+def test_run_cmd_no_runtime_info(mpi_config, tmp_path):
+    config_file = tmp_path / "config.yaml"
+    del mpi_config["component"]["runtime_info"]
+    YAMLConfig.dump_dict(config_file, mpi_config)
+    with patch.object(Driver, "_validate", return_value=True):
+        driver = ConcreteDriver(config_file=config_file)
+    driver._config = driver._experiment_config["component"]
+    with raises(KeyError):
+        driver.run_cmd()
+
+
+def test_scheduler():
+    config_file = fixture_path("fruit_config.yaml")
+    with patch.object(Driver, "_validate", return_value=True):
+        concretedriver = ConcreteDriver(config_file=config_file)
+        # pylint: disable=pointless-statement
+        with patch.object(driver.JobScheduler, "get_scheduler") as get_scheduler:
+            concretedriver.scheduler
+    get_scheduler.assert_called_once_with({})
+
+
+@pytest.mark.parametrize("link_files", [True, False])
+def test_stage_files(tmp_path, link_files):
+    """
+    Tests that files from static or cycle-dependent sections of the config obj are being staged
+    (copied or linked) to the run directory.
+    """
+
+    run_directory = tmp_path / "run"
+    src_directory = tmp_path / "src"
+    files_to_stage = yaml.safe_load(
+        """
+foo.yaml: some/foo/on/disk/foo.yml
+bar: somewhere/bar.txt
+./:
+  - a/file/with/same_name.x
+  - another/file/with/same/name.f
+"""
+    )
+    # Fix source paths so that they are relative to our test temp directory and
+    # create the test files.
+    src_directory.mkdir()
+    for dst_fn, src_path in files_to_stage.items():
+        if isinstance(src_path, list):
+            files_to_stage[dst_fn] = [str(src_directory / Path(sp).name) for sp in src_path]
+            for src_file in files_to_stage[dst_fn]:
+                Path(src_file).touch()
+        else:
+            fixed_src_path = src_directory / Path(src_path).name
+            files_to_stage[dst_fn] = str(fixed_src_path)
+            fixed_src_path.touch()
+    # Test that none of the destination files exist yet:
+    for dst_fn in files_to_stage.keys():
+        assert not (run_directory / dst_fn).is_file()
+    # Ask a forecast object to stage the files to the run directory:
```

Comment:
Is `forecast object` accurate in this comment? It looks to me like we're just calling `Driver` static methods.